cpp/src/structure/graph_impl.cuh

```diff
@@ -469,202 +473,239 @@ update_local_sorted_unique_edge_majors_minors(
+      local_sorted_unique_edge_minor_vertex_partition_offsets = std::move(h_key_offsets);
+    }
+  }
+
+  // 2. Update local_sorted_unique_edge_majors & local_sorted_unique_edge_major_offsets
+
+  std::vector<vertex_t> num_local_unique_edge_major_counts(edge_partition_offsets.size());
   for (size_t i = 0; i < edge_partition_offsets.size(); ++i) {
-    num_local_unique_edge_majors += thrust::count_if(
+    num_local_unique_edge_major_counts[i] += thrust::count_if(
       handle.get_thrust_policy(),
       thrust::make_counting_iterator(vertex_t{0}),
       thrust::make_counting_iterator(static_cast<vertex_t>(edge_partition_offsets[i].size() - 1)),
       has_nzd_t<vertex_t, edge_t>{edge_partition_offsets[i].data(), vertex_t{0}});
   }
-
-  auto [minor_range_first, minor_range_last] = meta.partition.local_edge_partition_minor_range();
-  rmm::device_uvector<uint32_t> minor_bitmaps(
-    ((minor_range_last - minor_range_first) + sizeof(uint32_t) * 8 - 1) / (sizeof(uint32_t) * 8),
-    handle.get_stream());
-  thrust::fill(handle.get_thrust_policy(), minor_bitmaps.begin(), minor_bitmaps.end(), uint32_t{0});
-  for (size_t i = 0; i < edge_partition_indices.size(); ++i) {
-    thrust::for_each(handle.get_thrust_policy(),
-                     edge_partition_indices[i].begin(),
-                     edge_partition_indices[i].end(),
-                     atomic_or_bitmap_t<vertex_t>{minor_bitmaps.data(), minor_range_first});
-  }
-
-  auto count_first = thrust::make_transform_iterator(minor_bitmaps.begin(), popc_t<vertex_t>{});
-  auto num_local_unique_edge_minors = thrust::reduce(
-    handle.get_thrust_policy(), count_first, count_first + minor_bitmaps.size(), vertex_t{0});
-
-  minor_bitmaps.resize(0, handle.get_stream());
-  minor_bitmaps.shrink_to_fit(handle.get_stream());
+  auto num_local_unique_edge_majors = std::reduce(num_local_unique_edge_major_counts.begin(),
+                                                  num_local_unique_edge_major_counts.end());
 
   vertex_t aggregate_major_range_size{0};
   for (size_t i = 0; i < meta.partition.number_of_local_edge_partitions(); ++i) {
     aggregate_major_range_size += meta.partition.local_edge_partition_major_range_size(i);
   }
-  auto minor_size = meta.partition.local_edge_partition_minor_range_size();
+
   auto max_major_properties_fill_ratio =
     host_scalar_allreduce(comm,
                           static_cast<double>(num_local_unique_edge_majors) /
                             static_cast<double>(aggregate_major_range_size),
                           raft::comms::op_t::MAX,
                           handle.get_stream());
-  auto max_minor_properties_fill_ratio = host_scalar_allreduce(
-    comm,
-    static_cast<double>(num_local_unique_edge_minors) / static_cast<double>(minor_size),
-    raft::comms::op_t::MAX,
-    handle.get_stream());
+  std::cout << "max_major_properties_fill_ratio=" << max_major_properties_fill_ratio
```

Comment:
Residual debug statement?

Reply:
Oops, I though I deleted this but it seems like I didn't :-) I will delete this.